src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -1134,56 +1188,117 @@ private void RunLayoutGraph(GraphLayout.Graph graph)
             graph.SetGraphPosition();
         }
 
-        private void SaveLayoutGraph(GraphLayout.Graph graph)
+        private void AvoidSubgraphOverlap()
+        {
+            var VerticalGraphDistance = 30;
+            var HorizontalGraphDistance = 70;
+
+            bool done;
+
+            do
+            {
+                done = true;
+
+                foreach (var g1 in Model.LayoutSubgraphs.Skip(1))
+                {
+                    foreach (var g2 in Model.LayoutSubgraphs.Skip(1))
+                    {
+                        if (!g1.Equals(g2) && (g1.GraphCenterY + g1.OffsetY <= g2.GraphCenterY + g2.OffsetY))
+                        {
+                            var g1nodes = g1.Nodes.OrderBy(n => n.Y + n.Height);
+                            var g2nodes = g2.Nodes.OrderBy(n => n.Y);
+
+                            foreach (var node1 in g1nodes)
+                            {
+                                foreach (var node2 in g2nodes)
+                                {
+                                    if ((node1.Y + node1.Height + VerticalGraphDistance + g1.OffsetY > node2.Y + g2.OffsetY) &&
+                                        (((node1.X <= node2.X) && (node1.X + node1.Width + HorizontalGraphDistance > node2.X)) ||
+                                        ((node2.X <= node1.X) && (node2.X + node2.Width + HorizontalGraphDistance > node1.X))))
+                                    {
+                                        g1.OffsetY -= 5;
+                                        g2.OffsetY += 5;
+                                        done = false;
+                                    }
+                                    if (!done) break;
+                                }
+                                if (!done) break;
+                            }
+                        }
+                    }
+                }
+            } while (!done);
+        }
```

Comment:
It appears that you're checking for overlap between each node in each subgraph. Is that right? Is it not possible to simply check for the bounds of each subgraph with all other subgraphs?